src/voice-broadcast/models/VoiceBroadcastPlayback.ts

```diff
@@ -150,23 +169,19 @@ export class VoiceBroadcastPlayback
         this.setInfoState(state);
     };
 
-    private async loadChunks(): Promise<void> {
-        const relations = getReferenceRelationsForEvent(this.infoEvent, EventType.RoomMessage, this.client);
-        const chunkEvents = relations?.getRelations();
-
-        if (!chunkEvents) {
-            return;
-        }
+    private async enqueueChunks(): Promise<void> {
+        const promises: Promise<void>[] = [];
 
-        this.chunkEvents.addEvents(chunkEvents);
-        this.setDuration(this.chunkEvents.getLength());
+        this.chunkEvents.getEvents().forEach((event: MatrixEvent) => {
```

Comment:
It can be simplified with:

```ts
const promises = this.chunkEvents.getEvents()
                              .filter(event => !this.playbacks.has(event.getId() || ""))
                              .map(event => this.enqueueChunk(event))
```

Reply:
Better to use `reduce` here. Otherwise there will be two iterations `filter` and `map`.

Reply:
Imo, it's more clear to filter and map than a reduce. It's two iterations of O(n)